libraries/connectivity/metrics/coherency.h

```diff
@@ -120,6 +120,28 @@ class CONNECTIVITYSHARED_EXPORT Coherency : public AbstractMetric
     static QVector<Eigen::MatrixXcd> computeCoherency(const QList<Eigen::MatrixXd> &matDataList,
                                                       int iNfft=-1,
                                                       const QString &sWindowType="hanning");
+
+private:
+    //=========================================================================================================
+    /**
+    * Computes the coherency values. This function gets called in parallel.
+    *
+    * @param[in] data    The input data.
+    *
+    * @return            The coherency result in form of AbstractMetricResultData.
+    */
+    static AbstractMetricResultData compute(const AbstractMetricInputData& data);
+
+    //=========================================================================================================
+    /**
+    * Reduces the coherency computation results to a final result. This function gets called in parallel.
+    *
+    * @param[out] finalData    The final data data.
+    * @param[in]  resultData   The resulting data from the computation step.
+    */
+    static void reduce(AbstractMetricResultData &finalData,
```

Comment:
Why is finalData not the return value?

Reply:
this is a requirement made by the QtConncurrent::mappedReduced function